### PR TITLE
weak tud_descriptor_string_cb and adjuest epin and epout

### PR DESCRIPTION
**Pull request by Pillar1989**

### CODE REVIEW

File: Adafruit_USBD_Device.cpp [hunk 1]
```diff
@@ -403,6 +403,8 @@ uint16_t const* tud_descriptor_string_cb(uint8_t index, uint16_t langid)
 TU_ATTR_WEAK uint8_t const * tud_hid_descriptor_report_cb(void) { return NULL; }
 TU_ATTR_WEAK uint16_t tud_hid_get_report_cb(uint8_t report_id, hid_report_type_t report_type, uint8_t* buffer, uint16_t reqlen) { return 0; }
 TU_ATTR_WEAK void tud_hid_set_report_cb(uint8_t report_id, hid_report_type_t report_type, uint8_t const* buffer, uint16_t bufsize) { }
+TU_ATTR_WEAK uint16_t const* tud_descriptor_string_cb(uint8_t index, uint16_t langid);
```
Comment (hathach) [inline Adafruit_USBD_Device.cpp:406]: not needed since it is implemented already.

File: tinyusb/src/device/usbd.h [hunk 2]
```diff
@@ -260,10 +260,11 @@ TU_ATTR_WEAK bool tud_vendor_control_complete_cb(uint8_t rhport, tusb_control_re
   9, TUSB_DESC_INTERFACE, _itfnum, 0, 2, TUSB_CLASS_HID, (uint8_t)((_boot_protocol) ? HID_SUBCLASS_BOOT : 0), _boot_protocol, _stridx,\
   /* HID descriptor */\
   9, HID_DESC_TYPE_HID, U16_TO_U8S_LE(0x0111), 0, 1, HID_DESC_TYPE_REPORT, U16_TO_U8S_LE(_report_desc_len),\
-  /* Endpoint Out */\
-  7, TUSB_DESC_ENDPOINT, _epout, TUSB_XFER_INTERRUPT, U16_TO_U8S_LE(_epsize), _ep_interval, \
   /* Endpoint In */\
-  7, TUSB_DESC_ENDPOINT, _epin, TUSB_XFER_INTERRUPT, U16_TO_U8S_LE(_epsize), _ep_interval
+  7, TUSB_DESC_ENDPOINT, _epin, TUSB_XFER_INTERRUPT, U16_TO_U8S_LE(_epsize), _ep_interval,\
```
Comment (hathach) [inline tinyusb/src/device/usbd.h:264]: why would you want to change the order ?

File: Adafruit_USBD_CDC.h [hunk 3]
```diff
@@ -40,6 +40,8 @@ class Adafruit_USBD_CDC : public Stream, Adafruit_USBD_Interface
 	void begin(uint32_t baud_count);
 	void begin(uint32_t baud, uint8_t config);
 	void end(void);
+        uint32_t getBaud() { return _baud_rate; }
+        void setBaud(uint32_t baud) {  _baud_rate = baud; }
```
Comment (hathach) [inline Adafruit_USBD_CDC.h:44]: These aren't Arduino std AP, it got ignored anyway. I don't see why we would need it

**Review by hathach — CHANGES_REQUESTED** [review]
weak callback declared is not needed, and I don't see why we need to change the order of EPs

**Comment by Pillar1989** [comment]
Sorry to bother you, but I'll do it myself.

**Pull request closed without merging**